src/engine-actions/close-channels.js

```diff
@@ -0,0 +1,83 @@
+const {
+  listChannels,
+  closeChannel,
+  listPendingChannels
+} = require('../lnd-actions')
+/**
+ * Closes a channel given the channel point
+ *
+ * @param {String} paymentChannelNetworkAddress
+ * @param {String} fundingAmount - int64 string
+ * @returns {Promise<void>} returns void on success
+ */
+async function closeChannels ({ force = false, pending = false } = {}) {
+  const { channels = [] } = await listChannels({ client: this.client })
+  const activeChannels = channels.filter(c => c.active)
+
+  let inactiveChannels = []
+  if (force === true) {
+    inactiveChannels = channels.filter(c => c.active === false)
+  }
+
+  let pendingChannels = []
+  if (pending === true) {
+    const { pendingOpenChannels = [] } = await listPendingChannels({ client: this.client })
+    pendingChannels = pendingOpenChannels
+  }
+  const closedChannelResponses = await Promise.all(activeChannels.concat(inactiveChannels).map(channel => close(channel, this.client, this.logger)))
+  const closedPendingChannelResponses = await Promise.all(pendingChannels.map(channel => close(channel.channel, this.client, this.logger)))
+
+  const allClosedChannels = closedChannelResponses.concat(closedPendingChannelResponses)
+  this.logger.info('Successfully closed channels', allClosedChannels)
+  return allClosedChannels
+}
+
+async function close (channel, client, logger) {
+  return new Promise((resolve, reject) => {
+    try {
+      const [fundingTxidStr, outputIndex] = channel.channelPoint.split(':')
+      const channelPoint = {
+        fundingTxidStr,
+        outputIndex: parseInt(outputIndex)
+      }
+      const close = closeChannel(channelPoint, false, { client })
+
+      const errorListener = (err) => {
+        logger.error(`Error from closeChannel stream`, err)
+        return finish(err)
+      }
+
+      const endListener = () => {
+        const error = `LND closed closeChannel stream before returning our value`
```

Comment:
[nit] backticks should fail `standardjs` because we are not using interpolation. We also don't need this variable since the line length isn't that large for `logger.error`

Reply:
I'm going to leave the variable because it's used in two places.